R/grade_result.R

```diff
@@ -62,3 +49,30 @@ grade_result <- function(
     message = message
   ))
 }
+
+#' Find a matching condition
+#' 
+#' Looks through a given set of conditions to attempt to find a match. 
+#' @param ... [pass_if()] or [fail_if()] [condition()]s
+#' @inheritParams grade_code
+#' @return The first condition that matches. If no conditional matches, returns 
+#' `NULL`.
+#' @export
+find_first_condition <- function(..., grader_args, learnr_args){
+  
+  results <- list(...)
+  chkm8_item_class(results, "grader_condition")
+  
+  if (!any(vapply(results, `[[`, logical(1), "correct"))) {
+    stop("At least one correct result must be provided")
+  }
```

Comment:
@trestletech if you don't mind, maybe we should take on https://github.com/rstudio-education/gradethis/issues/118 as a part of this PR, which *I think* would mean the following change:

```suggestion
  if (!any(vapply(results, `[[`, logical(1), "correct"))) {
    results <- c(results, list(pass_if(TRUE)))
  }
```